File: Data Structures & Algorithms/anagram-groups/submission-2.go
func groupAnagrams(strs []string) [][]string {
    anagramGroups := make(map[string][]string)
    for _, str := range strs {
        bytes := []byte(str)
        sort.Slice(bytes, func(i, j int) bool {
            return bytes[i] < bytes[j]
        })
        key := string(bytes)
        anagramGroups[key] = append(anagramGroups[key], str)
    }
    result := make([][]string, 0)
    for _, grp := range anagramGroups {
        result = append(result, grp)
    }
    return result
}

File: Data Structures & Algorithms/three-integer-sum/submission-2.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        three_sum = []
        n = len(nums)
        nums.sort()
        for i in range(n-2):
            j, k = i+1, n-1
            while j < k:
                s = nums[i]+nums[j]+nums[k]
                if s > 0:
                    k -= 1
                elif s < 0:
                    j += 1
                else:
                    if [nums[i], nums[j], nums[k]] not in three_sum:
                        three_sum.append([nums[i], nums[j], nums[k]])
                    k -= 1
                    
        return list(three_sum)
        